src/components/Cauldrons.vue

```diff
@@ -0,0 +1,197 @@
+    <table class="text-light">
+      <tr class="border-bottom">
+        <th class="centered columnSpan"
+          v-for="(item, index) in headers" :key="index">
+          {{item}}
+        </th>
+        <th>Materials</th>
+      </tr>
+      <tr class="alter-color"
+        v-for="n in 15" :key="n">
+        <AlchemyRow @customChange="handleDiscountRecalc()" :group="activeGroup" :idx="amountBubbles-n" :discount="discount" />
+      </tr>
+    </table>
+  </div>
+</template>
+
+<script lang="ts">
+  import { computed, defineComponent, ref } from "vue";
+  import { useState} from "~/State";
+  import AlchemyRow from "~/components/AlchemyRow.vue";
+  import { Color, Alch } from "~/composables/Alchemy";
+
+
+  export default defineComponent({
+      name: "Cauldrons",
+      components: {
+          AlchemyRow,
+      },
+      setup() {
+          const state = useState();
+          const alchemy = computed({
+            get: () => state.value.alchemy,
+            set: (value) => (state.value.alchemy = value),
+          });
+
+          const colors: Color[] = ["Orange", "Green", "Purple", "Yellow"];
+          const colorScheme = ["#ff9420", "#47db5a", "#cb79e3", "#edc300"];
+          const naming = ["Power", "Quicc", "High IQ", "Kazam"];
+
+          const headers = ["Icon", "Level", "Goal", "Effect"];
+          const amountBubbles = 15;
+
+          const activeGroup = ref("Orange"); 
+          const cauldRedux = ref(0);
+          const bargainLvl = ref(0);
+          let discount = ref([0,0,0,0,0]);
+
+          // Handler for discount calculations
+          const handleDiscountRecalc = () => {
+            let twelveBubbleLevel = alchemy.value.upgrades[activeGroup.value][14];
+            let undevLevel = alchemy.value.upgrades["Yellow"][6];
+            let vialLevel = alchemy.value.vials["Barley Brew"] ?? 0;
+            let discountMulti = Alch.discount(
+                Number(cauldRedux.value), 
+                undevLevel, vialLevel, 
+                twelveBubbleLevel, 
+                Number(bargainLvl.value));
+
+            Object.assign(discount.value, discountMulti);
+          };
+          
+          // TODO: I have the feeling this can be solved differently with some sort of enum system. Not sure
+          //        how those  work in JS/Vue though.
+          const colorTo = (color: Color, option: string) => {
+            let idx = colors.indexOf(color);
+            switch(option) {
+              case "Name":
+                return naming[idx];
+              case "Hex":
+                return colorScheme[idx];
+            }
+          };
+
+
+          return {      
+            handleDiscountRecalc,
+            colorTo,
+            discount,
+            alchemy,
+            colors,
+            activeGroup,
+            cauldRedux,
+            bargainLvl,
+            headers,
+            amountBubbles,
+          };
+      },
+      methods: {
+          setGroup(tab: string) {
```

Comment:
```suggestion
          setGroup(tab: Color) {
```
`Tab` is always a `Color`. By changing this and the `activeGroup` to `Color` all type errors are fixed.